mlflow/pipelines/steps/ingest/__init__.py

```diff
@@ -77,6 +93,21 @@ def _run(self, output_directory: str) -> BaseCard:
         _logger.debug("Successfully stored data in parquet format at '%s'", dataset_dst_path)
 
         ingested_df = read_parquet_as_pandas_df(data_parquet_path=dataset_dst_path)
+        if self.step_class() == StepClass.TRAINING:
+            if self.target_col not in ingested_df.columns:
+                raise MlflowException(
+                    f"Target column '{self.target_col}' not found in ingested dataset.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+            if self.positive_class is not None:
+                cardinality = ingested_df.apply(pd.Series.nunique)[self.target_col]
```

Comment:
```suggestion
                cardinality = ingested_df[self.target_col].nunique()
```

Can we only apply `nunique` to the target column?


### Benchmark:

```python
import numpy as np
import pandas as pd
import timeit

df = pd.DataFrame(np.random.randint(0, 10, size=(10000, 5)), columns=list("ABCDE"))


def subscript_then_nunique():
    return df["A"].nunique()


def nunique_then_subscript():
    return df.apply(pd.Series.nunique)["A"]


assert subscript_then_unique() == unique_then_subscript()

print(timeit.timeit(subscript_then_nunique, number=10000), "seconds")
print(timeit.timeit(nunique_then_subscript, number=10000), "seconds")
```

Output

```
0.542079824999746 seconds
4.465884477000145 seconds
```